src/p2p/swarm.rs

```diff
@@ -242,8 +243,12 @@ impl NetworkBehaviour for SwarmApi {
     fn poll(
         &mut self,
         _: &mut Context,
-        _: &mut impl PollParameters,
+        params: &mut impl PollParameters,
     ) -> Poll<NetworkBehaviourAction> {
+        if params.supported_protocols().len() != self.protocols.len() {
```

Comment:
`PollParameters::supported_protocols` returns an `ExactSizeIterator`, so obtaining the `len` is cheap